src/platform/silabs/MigrationManager.cpp

```diff
@@ -132,6 +170,73 @@ void MigrateHardwareVersion(void)
+    uint32_t secondPageAddr = tokenStartAddr + FLASH_PAGE_SIZE;
+    uint32_t credsBaseAddr  = 0;
+
+    // when credentials have been provided and stored in the first page of the static token location, we temporarely migrate them to
+    // the second page.
+    if (CHIP_NO_ERROR == SilabsConfig::ReadConfigValue(SilabsConfig::kConfigKey_Creds_Base_Addr, credsBaseAddr) &&
+        (credsBaseAddr >= tokenStartAddr && credsBaseAddr < secondPageAddr))
+    {
+        uint32_t cdSize                = 0;
+        uint32_t dacSize               = 0;
+        uint32_t paiSize               = 0;
+        bool actionSucceed             = false;
+        Provision::Manager & provision = Provision::Manager::GetInstance();
+
+        // Read the size of each credential type to determine the buffer size needed
+        actionSucceed = (SilabsConfig::ReadConfigValue(SilabsConfig::kConfigKey_Creds_CD_Size, cdSize) == CHIP_NO_ERROR);
+        actionSucceed &= (SilabsConfig::ReadConfigValue(SilabsConfig::kConfigKey_Creds_DAC_Size, dacSize) == CHIP_NO_ERROR);
+        actionSucceed &= (SilabsConfig::ReadConfigValue(SilabsConfig::kConfigKey_Creds_PAI_Size, paiSize) == CHIP_NO_ERROR);
+
+        if (actionSucceed)
+        {
+            // Depending on existing configuration, certifications could be overlapping from the first page to the second page.
+            // To mitigate any risk, we want to read all buffer before starting to move any of them.
+            // allocate buffers for each certificate.
+            uint8_t * dacBuffer = new uint8_t[dacSize];
+            uint8_t * paiBuffer = new uint8_t[paiSize];
+            uint8_t * cdBuffer  = new uint8_t[cdSize];
+            actionSucceed       = (dacBuffer != nullptr && paiBuffer != nullptr && cdBuffer != nullptr);
+            MutableByteSpan dacBufferSpan(dacBuffer, dacSize);
+            MutableByteSpan paiBufferSpan(paiBuffer, paiSize);
+            MutableByteSpan cdBufferSpan(cdBuffer, cdSize);
+
+            // Step to read the certificates at their current location
+            if (actionSucceed)
+            {
+                provision.Init();
+                // Read all certs and store it in our allocated buffer
+                actionSucceed = (provision.GetStorage().GetDeviceAttestationCert(dacBufferSpan) == CHIP_NO_ERROR);
+                actionSucceed &= (provision.GetStorage().GetProductAttestationIntermediateCert(paiBufferSpan) == CHIP_NO_ERROR);
+                actionSucceed &= (provision.GetStorage().GetCertificationDeclaration(cdBufferSpan) == CHIP_NO_ERROR);
+            }
+
+            // Step to write the certificates to their new location
+            if (actionSucceed)
+            {
+                provision.GetStorage().Initialize(0, 0);
+                provision.GetStorage().SetCredentialsBaseAddress(secondPageAddr);
+                // Write all certs back to the second page
+                // On the first write, we erase the new page.
```

Comment:
where is the erase done?

Reply:
In the function under the comment on the first write after Initialize